src/sentry/lang/native/symbolicator.py

```diff
@@ -420,3 +428,32 @@ def query_task(self, task_id):
 
     def reset_worker_id(self):
         self.worker_id = uuid.uuid4().hex
+
+
+class _Backoff:
+    """
+    Creates a new exponential backoff.
+    """
+
+    def __init__(self, initial, max):
+        """
+        :param initial: The initial backoff time in seconds.
+        :param max: The maximum backoff time in seconds.
+        """
+        self.initial = initial
+        self.max = max
+        self._current = 0
+
+    def reset(self):
+        """
+        Resets the backoff time zero.
+        """
+        self._current = 0
+
+    def sleep_failure(self):
+        """
+        Sleeps until the next retry attempt and increases the backoff time for the next failure.
+        """
+        if self._current > 0:
+            time.sleep(self._current)
+        self._current = min(max(self._current * 2, self.initial), self.max)
```

Comment:
A unit test would be nice here. E.g. monkey-patch `time.sleep` and then assert it gets called with `0.1, 0.2, 0.4, 0.8, 1.6, .3.2, 5, 5`.